Userland/Libraries/LibProtocol/WebSocketClient.h

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2021, The SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/HashMap.h>
+#include <LibIPC/ServerConnection.h>
+#include <WebSocket/WebSocketClientEndpoint.h>
+#include <WebSocket/WebSocketServerEndpoint.h>
+
+namespace Protocol {
+
+class WebSocket;
+
+class WebSocketClient
+    : public IPC::ServerConnection<WebSocketClientEndpoint, WebSocketServerEndpoint>
+    , public WebSocketClientEndpoint {
+    C_OBJECT(WebSocketClient);
+
+public:
+    virtual void handshake() override;
+
+    RefPtr<WebSocket> connect(const URL& url, const String& origin = {}, const Vector<String>& protocols = {}, const Vector<String>& extensions = {}, const HashMap<String, String>& request_headers = {});
```

Comment:
The parameter name `url` is implied by the type (`URL`) here and should be omitted per the project coding style.